tests/generation/test_utils.py

```diff
@@ -1599,18 +1609,27 @@ def test_assisted_decoding_sample(self):
             config.use_cache = True
             config.is_decoder = True
```

Comment:
Can we also have a test for when `config.is_encoder_decoder` to make sure any relevant logic is handled there?